libvast/vast/system/status.hpp

```diff
@@ -0,0 +1,130 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "vast/fwd.hpp"
+
+#include "vast/defaults.hpp"
+#include "vast/logger.hpp"
+#include "vast/system/actors.hpp"
+
+#include <caf/duration.hpp>
+#include <caf/settings.hpp>
+#include <caf/typed_response_promise.hpp>
+
+namespace vast::system {
+
+/// The verbosity level of a status request.
+enum class status_verbosity { info, detailed, debug };
+
+template <class Ptr, class Result>
+struct status_request_state_base {
+  // The actor handling the original request.
+  Ptr self;
+  // Promise to the original request.
+  caf::typed_response_promise<Result> promise;
```

Comment:
We need to be careful here not to access the promise outside of the context of the actor that created it. Can't we just call `self->make_response_promise<Result>` in the wrapping code and remove it from this state?

Reply:
The promise is part of the request state object because their lifetime is intentionally coupled. Would you be satisfied if `status_request_state_base` was not copyable?

Reply:
Yes, that would help a lot.